src/components/KeyboardShortcuts/index.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+
+import styles from './styles.module.scss';
+import { gettext } from '../../utils';
+
+const keys = ['k', 'j', 'e', 'h'];
+
+type Props = {};
+
+export class KeyboardShortcutsBase extends React.Component<Props> {
+  keydownListener = (event: KeyboardEvent) => {
```

Comment:
I believe React gives us a synthetic keyboard event so the type should reflect that: https://reactjs.org/docs/events.html#keyboard-events

Reply:
What I have gathered from the documentation and my observation, is that we are not receiving React synthetic keyboard events, but rather native keyboard events. I believe that if we use an `onKeyDown` attribute on a React element then React does generate synthetic keyboard events, but in this case we are adding an event listener to the `document` object and therefore we seem to be getting native keyboard events.

If I try to use `React.KeyboardEvent` or `React.KeyboardEvent<HTMLDivElement>` instead of `KeyboardEvent` then TS gets mad about me attempting to use `this.keydownListener` in my call to `document.addEventListener`:

```
Argument of type '(event: KeyboardEvent<HTMLDivElement>) => void' is not assignable to parameter of type 'EventListenerOrEventListenerObject'.
  Type '(event: KeyboardEvent<HTMLDivElement>) => void' is not assignable to type 'EventListener'.
    Types of parameters 'event' and 'evt' are incompatible.
      Type 'Event' is missing the following properties from type 'KeyboardEvent<HTMLDivElement>': altKey, charCode, ctrlKey, getModifierState, and 12 more.ts(2345)
```

I could be wrong about this, but from what I can tell `KeyboardEvent` is correct and `React.KeyboardEvent` is not.

Reply:
>  but in this case we are adding an event listener to the `document` object and therefore we seem to be getting native keyboard events.

exactly